k2/csrc/ragged.h

```diff
@@ -212,6 +216,59 @@ class RaggedShape {
   std::vector<RaggedShapeLayer> layers_;
 };
 
+
+template <typename T, int MAX_DIM>
+struct ArrayAccessor {
+  T data[MAX_DIM];
+  ArrayAccessor() { }
```

Comment:
I would suggest removing the default constructor and the copy constructor.
The default behavior provided by the compiler is good enough for this case.

The struct becomes a non-POD type after adding them.

Reply:
I needed to add it because it needs a __host__ __device__ copy constructor (to be transferable to GPU) but if that exists then it doesn't get a default constructor.

Reply:
I don't believe a __device__ copy constructor gets added by default?  (May be wrong though.)

Reply:
oh you're right, it seems to still work after removing them.

Reply:
BTW,

```cpp
ArrayAccessor() { }
```
and
```cpp
ArrayAccessor() = default;
```

are not the same. I would recommend the second approach (using `=default`) if you really want to add it.

Reply:
```cpp
template <typename T, int MAX_DIM>
struct ArrayAccessor {
  T data[MAX_DIM];
};

void TestArrayAccessor() {
  ContextPtr c = GetCudaContext();

  ArrayAccessor<int32_t, 2> acc;
  acc.data[0] = 10;
  acc.data[1] = 20;
  Array1<int32_t> arr(c, 2);
  int32_t *arr_data = arr.Data();
  K2_EVAL(
      c, 2, lamda_set, (int32_t i) { arr_data[i] = acc.data[i]; });
  K2_LOG(INFO) << arr;
}
```

prints `[10, 20]`